course/auth.py

```diff
@@ -270,6 +270,24 @@ def stop_impersonating(request):
         raise PermissionDenied()
 
     if not hasattr(request, "relate_impersonate_original_user"):
+        # prevent user without pperm to stop_impersonating
+        my_participations = Participation.objects.filter(
+            user=request.user,
+            status=participation_status.active)
```

Comment:
So `request.user` is the impersonated user, which means that getting out of impersonating someone now would require a logout?

Reply:
No, it was preventing who didn't currently impersonating and who don't have impersonating pperm to do `stop_impersonating`. You can see from the testcase [here](https://github.com/inducer/relate/pull/388/files#diff-b4c9abe346f4b3686dd3e1025a50aaa7R116)

Reply:
Without this, student user can view the `impersonate` page. This patch redirects them to 403.

Reply:
I know that requires extra db hits when stop impersonating, but I just want them to know there are permissions.

Reply:
> This patch redirects them to 403.

The tests is (here)[https://github.com/inducer/relate/pull/388/files#diff-b4c9abe346f4b3686dd3e1025a50aaa7R81]. Without this, the `status_code` is 200(302).

Reply:
Sure. It doesn't hurt. Thanks for working on this!